### DISCUSSION TITLE
Netlist Viewer Icons

### DISCUSSION THREAD

**Discussion by Lramseyer** [Lramseyer/vaporview#23] [Design Opinions]
As the title suggests, I want to improve the netlist viewer icons. I can create custom icons, or use ones that are [provided by the VScode API](https://code.visualstudio.com/api/references/icons-in-labels). When I originally created the netlist viewer, I only planned for VCD files and the types that were supported. But I am in the process of adding FST support, and I want a more mature choice in icons and color scheme for said icons.

In essence, there are scope types (ie a module) and variable types (reg, wire, single bit, multi bit). I want to make it obvious which is which without making it confusing.

I'm curious to hear all of your thoughts and ideas on the matter!

**Comment by krisjdev** [comment]
I think some good candidates for showing scopes are ``symbol-constructor``, ``symbol-module`` and ``symbol-structure`` - maybe ``symbol-structure`` can be used for the top module? 

As for variable types, perhaps ``symbol-array``, ``symbol-variable`` or even ``symbol-boolean``? I'm not sure how you'd convey the width of the signal via icons if that's something you wanted to do - that would probably require a custom set.
